mono/utils/mono-rand.c

```diff
@@ -172,7 +183,11 @@ mono_rand_try_get_bytes (gpointer *handle, guchar *buffer, gint buffer_size, Mon
 		gint err;
 
 		do {
+#ifdef HAVE_GETRANDOM
```

Comment:
Can't we avoid opening `/dev/random` if getrandom is available?

Reply:
Would it be a good idea to separate all functions, if getrandom is available? Something like this in the case of getrandom?

```
gboolean
mono_rand_open (void)
{
	return TRUE;
}

gpointer
mono_rand_init (guchar *seed, gint seed_size)
{
	return "getrandom"; // NULL will be interpreted as failure; return arbitrary nonzero pointer
}

gboolean
mono_rand_try_get_bytes (gpointer *handle, guchar *buffer, gint buffer_size, MonoError *error)
{
	gint count = 0;
	gint err;

	error_init (error);

	/* Read until the buffer is filled. This may block if random pool isn't initialized. */
	do {
		err = getrandom (buffer + count, buffer_size - count, 0);
		if (err < 0) {
			if (errno == EINTR)
				continue;
			g_warning("Entropy error! Error in getrandom (%s).", strerror (errno));
			/* exception will be thrown in managed code */
			mono_error_set_execution_engine (error, "Entropy error! Error in getrandom (%s).", strerror (errno));
			return FALSE;
		}
		count += err;
	} while (count < buffer_size);
	return TRUE;
}
```